### PR TITLE
admin page and routes added

### PR DESCRIPTION
**Pull request by hamoudi-hamoudi**
## 📝 Description
added components for the admin plus routes for the back-end 
the home component have a condition to render different content if the the user_type is admin
CSS applied basically and functionality for see-all in JobApproval component yet to come. need to adjust the routing system first
all admin calls for back-end are in admin.store.ts 

## ⚠️ Notes

having a problem with CSS on deny or approve click .

## 🎉 Demos

Screenshot 2023-10-04 144955
Screenshot 2023-10-04 145031

### CODE REVIEW

File: frontend/src/pages/admin/AdminPage.tsx
Comment (BrantalikP) [inline frontend/src/pages/admin/AdminPage.tsx]: All those functions are asynchronous therefore they should be awaited, also I recommend extracting it to a separate function named `getAdminData` so its clear why we are calling all of those

File: frontend/src/pages/admin/AdminPage.tsx
Comment (BrantalikP) [inline frontend/src/pages/admin/AdminPage.tsx]: Are we going to use those values also outside of the AdminPage? If we wont we can simply call the fetch methods here and store it in the `useState` hook

Comment (BrantalikP) [reply frontend/src/pages/admin/AdminPage.tsx]: this is not crucial, just a global store as ZUSTAND should be used more for values that are needed across multiple screens and features (e.g theme, user data... )

File: frontend/src/pages/admin/AdminPage.tsx
Comment (BrantalikP) [reply frontend/src/pages/admin/AdminPage.tsx]: You can incorporate this logic into a separate hook named ﻿useGetAdminData within the admin folder. In this hook, you should call all the necessary methods and return their results to ﻿AdminPage. This might not seem significant now, but this decoupling of fetching and rendering processes can make your code more testable in the future and of course easier to read :D

File: frontend/src/components/adminDetails/jobApproval/JobApproval.tsx
Comment (BrantalikP) [inline frontend/src/components/adminDetails/jobApproval/JobApproval.tsx]: Capital N

File: frontend/src/components/adminDetails/jobApproval/JobApproval.tsx
Comment (BrantalikP) [inline frontend/src/components/adminDetails/jobApproval/JobApproval.tsx]: you know the type, its the Job interface from `jobsToApprove` you can export it and use it here as `jobs: Job[]`

File: frontend/src/components/adminDetails/jobApproval/JobApproval.tsx
Comment (BrantalikP) [inline frontend/src/components/adminDetails/jobApproval/JobApproval.tsx:8]: Why are we slicing the array?

Comment (BrantalikP) [reply frontend/src/components/adminDetails/jobApproval/JobApproval.tsx:8]: I see, my bad, you will extend it when the user clicks on "see all" button 👍

**Review by BrantalikP — DISMISSED** [review]
Hey @hamoudi-hamoudi great job! this is a huge step 👏 I have not found anything crucial the only thing is the design compliance what CSS issues are you experiencing? 

https://github.com/HackYourFuture-CPH/class-25-ironside/assets/48878547/6e40596f-a02c-49de-8c6d-a60e56ca6333

And if I understood correctly the UI is still in progress, right? Because the design is not good right now, we need to fix it. What about keeping this PR focused only on the functionality and doing the UI in a separate PR so we are not blocked?
Screenshot 2023-10-05 at
![Screenshot 2023-10-05 at 8 41 43
 8 30 15](https://github.com/HackYourFuture-CPH/class-25-ironside/assets/48878547/c5a31ac1-f6ec-4459-b02a-bd600982cd20)
Screenshot 2023-10-05 at 8 37 35
Screenshot 2023-10-05 at 8 39 24

File: frontend/src/pages/admin/AdminPage.tsx
Comment (BrantalikP) [reply frontend/src/pages/admin/AdminPage.tsx]: The benefits of using custom hooks includes keeping the DRY (Do Not Repeat Yourself) convention, which is particularly useful when you need to use the same logic across multiple modules. Custom hooks assist in separating UI components from functional elements, thereby enhancing code readability. Moreover, they improve testability, among other advantages. Let me illustrate these concepts with the ﻿`useAdminData` hook how we can incorporate it into the current project architecture.

1. create a folder `hooks` in the `src` directory
2. lets create a new file `useAdminData`
3. extract the logic from AdminPage and move it into the file
```
import { useEffect } from 'react';
import useAdminStore from '../store/admin.store';

export const useAdminData = () => {
  const jobsRequest = useAdminStore((state) => state.jobsRequest);
  const getJobRequest = useAdminStore((state) => state.getJobRequest);
  const usersCount = useAdminStore((state) => state.usersCount);
  const getUserCount = useAdminStore((state) => state.getUserCount);
  const companiesCount = useAdminStore((state) => state.companiesCount);
  const getComaniesCount = useAdminStore((state) => state.getComaniesCount);
  const jobsCount = useAdminStore((state) => state.jobsCount);
  const getJobsCount = useAdminStore((state) => state.getJobsCount);
  const openJobsCount = useAdminStore((state) => state.openJobsCount);
  const getOpenJobsCount = useAdminStore((state) => state.getOpenJobsCount);

  useEffect(() => {
    getJobRequest();
    getUserCount();
    getComaniesCount();
    getJobsCount();
    getOpenJobsCount();
  }, []);


  return { usersCount, companiesCount, jobsCount, jobsRequest, openJobsCount }
}
```
4. import useAdminData hook back into Admin page and use the data from it.
`const { usersCount, companiesCount, jobsCount, jobsRequest, openJobsCount } = useAdminData()
 `

Now whenever you need those data to be fetched or accessed you can simple re-use this hook 
PS: I am still not sure if we really need the keep the data in zustand storage

Comment (hamoudi-hamoudi) [reply frontend/src/pages/admin/AdminPage.tsx]: Great Idea , Thank you so much for explaining :heart: . I will make the changes

File: frontend/src/pages/admin/AdminPage.tsx
Comment (hamoudi-hamoudi) [reply frontend/src/pages/admin/AdminPage.tsx]: Honestly, we have been asked to use Zustand. I couldn't find another purpose for it.
I might use it in another pages in when we advance a little also thought its nice new technology for us to practice :grimacing:

File: frontend/src/components/adminDetails/jobApproval/JobApproval.tsx
Comment (hamoudi-hamoudi) [reply frontend/src/components/adminDetails/jobApproval/JobApproval.tsx]: On it :grin:

**Comment by hamoudi-hamoudi** [comment]
Mozilla Firefox-20231006-1220074_gif

I will make the CSS in another PR later , the issue I am having with replacing the missing spot after deny or approve, it happns when i added CSS transition {.fadeOut} to element to desapear nicely

File: frontend/src/store/admin.store.ts
Comment (YurijPaharev) [inline frontend/src/store/admin.store.ts]: Please keep interfaces with a capital letter

**Review by YurijPaharev — CHANGES_REQUESTED** [review]
Very well done. Please check the comments and remove some of the linting problems

**Review by YurijPaharev — APPROVED** [review]

**Pull request merged**
